TypeSystem/flow-analysis/reachability_switch_expression_A02_t03.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion switch statement: If `N` is a switch statement of the form
+/// `switch (E) {alternatives}` then:
+/// - Let `before(E) = before(N)`.
+/// - For each `C` in `alternatives` with statement body `S`:
+///   - If `C` is labelled let
+///     `before(S) = conservativeJoin(after(E), assignedIn(N), capturedIn(N))`
+///     otherwise let `before(S) = after(E)`.
+///  - If the cases are exhaustive, then let `after(N) = break(N)` otherwise let
+///    `after(N) = join(after(E), break(N))`.
+/// TODO (sgrekhov): there is no switch expressions in the flow analysis spec
+/// yet (April 2025). Update the assertion after spec update.
+///
+/// @description Checks that if `C` is not labelled then `before(S) = after(E)`.
+/// Test that if `C` is not labeled and a promoted variable was captured in `E`
+/// then it is demoted in `S`.
+/// @author sgrekhov22@gmail.com
+
+class C {
+  int v;
+  C(this.v);
+}
+
+void test1(int? n) {
+  if (n != null) {  // n promoted to int
+    var x = switch ((){n = 42;}) { // n demoted to int?
```

Comment:
It seems reasonable to avoid the "never_matches" diagnostic because this makes the code somewhat artificial (I don't have a really good reason, but it seems reasonable to focus mostly on code that could actually run and reach the pieces of code which are in focus for the test). So:
```suggestion
    var x = switch ((){n = 42; return 1;}) { // `n` demoted to `int?`.
```
Similarly for the other cases.

Reply:
A very good advice! Thank you! Updated.

Reply:
Thanks!